### DISCUSSION TITLE
LSP failing to find views declared in other files

### DISCUSSION THREAD

**Discussion by kiahjh** [wojciech-kulik/xcodebuild.nvim#173] [Q&A] [answered]
Hi! Absolutely loving this plugin, so nice to be out of xcode ❤️
Almost everything is working fine for me, but I was wondering if anyone has run into this issue where when I try to compose one view into another, LSP tells me it `cannot find 'SomeView' in scope`.

<img width="663" alt="Screenshot 2024-09-04 at 4 52 31 PM" src="https://github.com/user-attachments/assets/2f28359c-9972-4ac0-a939-b52a684eea68">

The issue only exists in xcode projects; using things declared in other files is working perfectly within SPM packages, and I'm having no other issues in those.

Here's my config:
```lua
lspconfig.sourcekit.setup({
  capabilities = capabilities,
  on_attach = on_attach,
  handlers = handlers,
  cmd = {
    "/usr/bin/sourcekit-lsp",
  },
  single_file_support = false, -- don't want this, because it falls back to the cwd if it can't find an xcode proj or spm package

  -- if it's an spm package (it has a Package.swift), then root_dir should be the root of the package; if it's an xcode project, then root_dir should be wherever buildServer.json is:
  root_dir = function(fname)
    return lspconfig.util.root_pattern("Package.swift", "buildServer.json")(fname)
      -- some reasonable falbacks:
      or lspconfig.util.find_git_ancestor(fname)
      or vim.fn.getcwd()
  end,
})
```

Everything is building and previewing fine, it's just a misconception of `sourcekit-lsp`. I've tried restarting the lsp server, quitting/reopening nvim, etc.

Thanks in advance!

**Comment by wojciech-kulik** [comment]
Hi! 

Could you switch to this config and see if it works:
https://github.com/wojciech-kulik/ios-dev-starter-nvim
?

Also, please check if `buildServer.json` contains the correct path and if `:LspInfo` show the correct root directory.
Additionally, you can try changing your `sourcekit-lsp` path to: `vim.trim(vim.fn.system("xcrun -f sourcekit-lsp"))`

**Reply by kiahjh** [reply]
Thanks for the reply! I made a new test repo and tried everything you suggested, but I'm still having the same issues. Here's the repo, in case you want to clone it down and poke around: https://github.com/kiahjh/sourcekit-in-xcode-test

I put a description of what I tried and the output of `:LspInfo` in the readme.

**Reply by wojciech-kulik** [reply]
I tried your projects and it works well on my side, no issues and I'm able to navigate to definitions:
<img width="650" alt="image" src="https://github.com/user-attachments/assets/220e6d21-c177-485a-8872-27daaae0c9ac">

**Reply by wojciech-kulik** [reply, marked as answer]
I see that your `xcode-build-server` is in `/Users/miciah/.local/bin/xcode-build-server`. Maybe try installing it via homebrew?

**Reply by kiahjh** [reply]
Deleting that copy of `xcode-build-server` and reinstalling it with homebrew totally fixed the issue. Thank you so much for the help!

**Comment by Woit** [comment]
xcodebuild using xcode-build-server for lsp interaction. Did you setup it? (via `brew install xcode-build-server` as example). If so, maybe you need build project once in xcode, because time to time xcode-build-server may have strange behaviour. I have the same issue every time, when i clearing DerivedData folder.
